distribution/docker/src/docker/bin/install-baseimage.sh

```diff
@@ -0,0 +1,149 @@
+#
+# Originally from:
+# 
+# https://github.com/moby/moby/blob/master/contrib/mkimage-yum.sh
+
+set -e
+
+usage() {
+    cat <<EOOPTS
+$(basename $0) <platform> <output_file>
+EOOPTS
+    exit 1
+}
+
+platform="$1"
+output_file="$2"
+
+if [[ -z $platform ]]; then
+    usage
+fi
+
+if [[ -z "$output_file" ]]; then
+    usage
+fi
+
+# Start off with an up-to-date system
+yum update --setopt=tsflags=nodocs -y
+
+# Create a temporary directory into which we will install files
+target=$(mktemp -d --tmpdir $(basename $0).XXXXXX)
+
+set -x
+
+# Create required devices
+mkdir -m 755 "$target"/dev
+mknod -m 600 "$target"/dev/console c 5 1
+mknod -m 600 "$target"/dev/initctl p
+mknod -m 666 "$target"/dev/full c 1 7
+mknod -m 666 "$target"/dev/null c 1 3
+mknod -m 666 "$target"/dev/ptmx c 5 2
+mknod -m 666 "$target"/dev/random c 1 8
+mknod -m 666 "$target"/dev/tty c 5 0
+mknod -m 666 "$target"/dev/tty0 c 4 0
+mknod -m 666 "$target"/dev/urandom c 1 9
+mknod -m 666 "$target"/dev/zero c 1 5
+
+# Install files. We attempt to install a headless Java distro, and exclude a
+# number of unnecessary dependencies. In so doing, we also filter out Java itself,
+# but since Elasticsearch ships its own JDK, with its own libs, that isn't a problem
+# and in fact is what we want.
+#
+# Note that we also skip coreutils, as it pulls in all kinds of stuff that
+# we don't want.
+#
+# Note that I haven't yet verified that these dependencies are, in fact, unnecessary.
+#
+# We also include some utilities that we ship with the image.
+#
+#   * `nc` is useful for checking network issues
+#   * `zip` and `unzip` are for working with bundles
+#   * `pigz` is used for compressing large heaps dumps, and is considerably faster than `gzip` for this task.
+#   * `tini` is a tiny but valid init for containers. This is used to cleanly control how ES and any child processes are shut down.
+#
+yum --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
+  --setopt=group_package_types=mandatory -y  \
+  -x copy-jdk-configs -x cups-libs -x javapackages-tools -x alsa-lib -x freetype -x libjpeg -x libjpeg-turbo \
+  -x coreutils \
+  --skip-broken \
+  install \
+    java-latest-openjdk-headless \
+    bash nc zip upzip pigz
+
+ARCH="$(basename $platform)"
+curl --retry 10 -L -o "$target"/bin/tini-static-$ARCH           "https://github.com/krallin/tini/releases/download/v0.19.0/tini-static-$ARCH"
+curl --retry 10 -L -o "$target"/bin/tini-static-$ARCH.sha256sum "https://github.com/krallin/tini/releases/download/v0.19.0/tini-static-$ARCH.sha256sum"
+(cd "$target/bin" && sha256sum -c tini-static-$ARCH.sha256sum)
+rm "$target"/bin/tini-static-$ARCH.sha256sum
+mv "$target"/bin/tini-static-$ARCH "$target"/bin/tini
+chmod +x "$target"/bin/tini
+
+# Use busybox instead of installing more RPMs, which can pull in all kinds of
+# stuff we don't want. There's no RPM for busybox available for CentOS.
+BUSYBOX_URL="https://busybox.net/downloads/binaries/1.31.0-i686-uclibc/busybox"
```

Comment:
At some point there will be bugs with these utilities (busybox, or tini) as well.

I think it's hard to keep track of the versions as they are hard coded here.

What if specify such versions as readonly vars at the top of this shell script so that it's easier to track the versions and do future updates?

example after
```
#!/usr/bin/env bash
...
```

then

```
declare -r BUSYBOX_VERSION="1.31.0"
declare -r TINI_VERSION="0.19.0"
```